src/GGTestUtils.py

```diff
@@ -333,10 +471,45 @@ def _upload_files_to_s3(self, files: Sequence[os.PathLike | str],
             print(
                 f"File {file_path} successfully uploaded to {bucket_name}/{object_name}"
             )
+
+        # Wait for S3 artifact propagation by checking availability
+        if files:
+            last_object = object_name    # Use the last uploaded file for checking
+            for attempt in range(20):    # Max 20 seconds
+                try:
+                    self._s3Client.head_object(Bucket=bucket_name,
+                                               Key=last_object)
+                    print(
+                        f"S3 artifact verified accessible after {attempt + 1}s")
+                    break
+                except:
+                    time.sleep(1)
+            else:
+                raise Exception(
+                    f"S3 artifact {bucket_name}/{last_object} not accessible after 20s"
+                )
+
+            # Verify ETag/digest is available for Greengrass validation
+            for attempt in range(10):
+                try:
+                    response = self._s3Client.head_object(Bucket=bucket_name, Key=last_object)
+                    if 'ETag' in response and response['ETag']:
+                        print(f"S3 artifact ETag verified after {attempt + 1}s")
+                        break
+                except:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Try, Except, Pass detected. https://bandit.readthedocs.io/en/latest/plugins/b110_try_except_pass.html